File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/main.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        white = 0
        for i in range(k):
            if blocks[i] == "W":
                white += 1
        res = white                
        for i in range(k, len(blocks)):
            if blocks[i] == "W":
                white += 1
            if blocks[i - k] == "W":
                white -= 1
            res = min(res, white)
        return res

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/main.go
package main

func minimumRecolors(blocks string, k int) int {
	white := 0
	for i := 0; i < k; i++ {
		if blocks[i] == 'W' {
			white++
		}
	}
	res := white
	for i := k; i < len(blocks); i++ {
		if blocks[i] == 'W' {
			white++
		}
		if blocks[i-k] == 'W' {
			white--
		}
		res = min(res, white)
	}
	return res
}
